tools/accuracy_checker/accuracy_checker/metrics/coco_orig_metrics.py

```diff
@@ -108,107 +63,144 @@ def parameters(cls):
-            map_pred_label_id_to_coco_cat_id = {k: coco_cat_name_to_id[v] for k, v in meta['label_map'].items()}
-        else:
-            map_pred_label_id_to_coco_cat_id = self.generate_map_pred_label_id_to_coco_cat_id(has_background,
-                                                                                              use_full_label_map)
-            for k, v in meta['label_map'].items():
-                assert map_pred_label_id_to_coco_cat_id[k] == coco_cat_name_to_id[v], (
-                    "k = {}, v = {}, map_pred_label_id_to_coco_cat_id[k] = {}, coco_cat_name_to_id[v] = {}".format(
-                        k, v, map_pred_label_id_to_coco_cat_id[k], coco_cat_name_to_id[v]))
+    @staticmethod
+    def _iou_type_data_to_coo(data_to_store, data):
+        x_mins = data.x_mins.tolist()
+        y_mins = data.y_mins.tolist()
+        x_maxs = data.x_maxs.tolist()
+        y_maxs = data.y_maxs.tolist()
+
+        for data_record, x_min, y_min, x_max, y_max in zip(
+                data_to_store, x_mins, y_mins, x_maxs, y_maxs
+        ):
+            width = x_max - x_min + 1
+            height = y_max - y_min + 1
+            area = width * height
+            data_record.update({
+                'bbox': [x_min, y_min, width, height],
+                'area': area
+            })
+
+        return data_to_store
+
+    def _prepare_coco_structures(self, annotations):
+        meta = self.dataset.metadata
 
-            assert all(map_pred_label_id_to_coco_cat_id[k] == coco_cat_name_to_id[v]
-                       for k, v in meta['label_map'].items())
+        if COCO is None:
+            raise ValueError('pycocotools is not installed, please install it')
+        coco_data_to_store = self._prepare_data_for_annotation_file(
+            annotations, meta.get('label_map'))
 
-        map_coco_img_file_name_to_img_id = {os.path.basename(v['file_name']): v['id'] for v in coco.dataset['images']}
-        assert len(map_coco_img_file_name_to_img_id) == len(coco.dataset['images']), "Image name duplications"
+        coco_annotation = self._create_json(coco_data_to_store)
+        coco = COCO(str(coco_annotation))
 
-        return coco, map_coco_img_file_name_to_img_id, map_pred_label_id_to_coco_cat_id
+        return coco
 
-    @staticmethod
-    def _convert_data_to_coco_format(
-            predictions, map_coco_img_file_name_to_img_id, map_pred_label_id_to_coco_cat_id, iou_type='bbox'
-    ):
+    def _convert_data_to_coco_format(self, predictions):
         coco_data_to_store = []
         for pred in predictions:
             prediction_data_to_store = []
-            cur_name = pred.identifier
-            cur_name = os.path.basename(cur_name)
-            assert cur_name in map_coco_img_file_name_to_img_id
-            cur_img_id = map_coco_img_file_name_to_img_id[cur_name]
+            cur_name = pathlib.PurePath(pred.identifier).name
+            cur_img_id = int(cur_name.split(".")[0])
```

Comment:
please pay attention that in COCO 2014  was another id naming policy

Reply:
please review again